src/invert/laplacexy/laplacexy.cxx

```diff
@@ -29,7 +29,8 @@ static PetscErrorCode laplacePCapply(PC pc,Vec x,Vec y) {
 }
 
 LaplaceXY::LaplaceXY(Mesh *m, Options *opt, const CELL_LOC loc)
-    : localmesh(m==nullptr ? bout::globals::mesh : m), location(loc) {
+    : lib(opt==nullptr ? &(Options::root()["laplacexy"]["petsc"]) : &(*opt)["petsc"]),
```

Comment:
Rather than storing a new petsc lib which has a specific options attached to it could we just store the ksp directly (and take an option prefix to use or similar)?

Reply:
Each of the solvers does directly store a `KSP` already. I guess it could make sense to have some sort of wrapper for `KSP` which would handle the `PetscLib` instances internally (or replace the functionality); would be on the way to the C++ interface to PETSc that Peter wants :-) but on the other hand, the options-handling can be abstracted away from whether the solver is a `KSP`, `SNES`, time-solver, etc. Without totally re-doing our PETSc interface, the `PetscLib` design (with methods to either create (as I've implemented at present), or (alternatively) set options for, `KSP`, `SNES`, etc. instances) doesn't seem to me like a bad compromise for now.

Reply:
Agreed, this is a reasonable compromise without needing to write a full C++ wrapper for PETSc